src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -285,47 +290,91 @@ protected override void StopProcessing()
 
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
         {
+            if(Repeat.IsPresent)
+            {
+                Count = int.MaxValue;
+            }
+            else if (!MyInvocation.BoundParameters.ContainsKey("Count"))
+            {
+                Count = 1;
+            }
+
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
+            int successfulConnections = 0;
 
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
-
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                TcpPortStatus testResult = new TcpPortStatus(
+                    i,
+                    Source,
+                    targetNameOrAddress,
+                    targetAddress.ToString(),
+                    TcpPort,
+                    0,
+                    TcpPortStatus.TcpConnectionTestResult.New
+                );
+
+                using(TcpClient client = new TcpClient())
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
-                    Task.WhenAny(connectionTask, timeoutTask).Result.Wait();
+                    try
+                    {
+                        Stopwatch stopwatch = new Stopwatch();
+                        stopwatch.Start();
```

Comment:
You can do this in a single line with the static `StartNew()` method:

```csharp
Stopwatch timer = Stopwatch.StartNew();
```

Reply:
Ah this is good to know! I've moved the stopwatch out of the loop to initialize it once, so we may not be able to add this in at this point. Definitely adding this to my notes though!

Reply:
StartNew() will do extra allocation.